JNWCollectionView/JNWCollectionViewFramework.m

```diff
@@ -131,9 +137,23 @@ - (id)initWithCoder:(NSCoder *)aDecoder {
 	return self;
 }
 
+#pragma mark Notifications
+
+- (void)preferredScrollerStyleDidChangeNotification:(NSNotification *)note {
+	NSView *documentView = self.documentView;
+	
+	if (!CGSizeEqualToSize(NSSizeToCGSize(documentView.frame.size), NSSizeToCGSize(self.contentSize))) {
+		CGRect documentFrame = documentView.frame;
+		documentFrame.size.width = self.contentSize.width;
+		documentView.frame = documentFrame;
```

Comment:
Apparently only setting the property to Overlay doesn't work, so glad you
found a better solution. Will try this tonight and will tell you tomorrow
if everything's alright in my side.

Thanks.
On Wed, 13 Jan 2016 at 08:10, Michał Zaborowski <notifications@github.com>
wrote:

> In JNWCollectionView/JNWCollectionViewFramework.m
> <https://github.com/jwilling/JNWCollectionView/pull/157#discussion_r49557321>
> :
>
> > @@ -131,9 +137,23 @@ - (id)initWithCoder:(NSCoder *)aDecoder {
> >  	return self;
> >  }
> >
> > +#pragma mark Notifications
> > +
> > +- (void)preferredScrollerStyleDidChangeNotification:(NSNotification *)note {
> > +	NSView *documentView = self.documentView;
> > +	
> > +	if (!CGSizeEqualToSize(NSSizeToCGSize(documentView.frame.size), NSSizeToCGSize(self.contentSize))) {
> > +		CGRect documentFrame = documentView.frame;
> > +		documentFrame.size.width = self.contentSize.width;
> > +		documentView.frame = documentFrame;
>
> I am iOS dev so i even didn't know about layoutDocumentView, in free time
> i will try replace it wigh layoutDocumentView, btw. cell frame are based on
> document view frame so anyway shouldn't invalidate layout be after layout
> document view?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/jwilling/JNWCollectionView/pull/157/files#r49557321>.
>